### ISSUE TITLE
Unable to install on CentOS 7.0

### ISSUE THREAD

**Issue by indigo423** [hw-cookbooks/postgresql#167]
[2014-07-26T01:46:08+00:00] INFO: package[postgresql-server] installing postgresql-server-9.2.7-1.el7 from base repository

           - install version 9.2.7-1.el7 of package postgresql-server

         * template[/etc/sysconfig/pgsql/postgresql] action create[2014-07-26T01:46:19+00:00] INFO: Processing template[/etc/sysconfig/pgsql/postgresql] action create (postgresql::server_redhat line 54)

           * Parent directory /etc/sysconfig/pgsql does not exist.
       ================================================================================
       Error executing action `create` on resource 'template[/etc/sysconfig/pgsql/postgresql]'
       ================================================================================

       Chef::Exceptions::EnclosingDirectoryDoesNotExist
       ------------------------------------------------
       Parent directory /etc/sysconfig/pgsql does not exist.

       Resource Declaration:
       ---------------------
       # In /tmp/kitchen/cookbooks/postgresql/recipes/server_redhat.rb

        54: template "/etc/sysconfig/pgsql/#{node['postgresql']['server']['service_name']}" do
        55:   source "pgsql.sysconfig.erb"
        56:   mode "0644"
        57:   notifies :restart, "service[postgresql]", :delayed
        58: end
        59:

       Compiled Resource:
       ------------------
       # Declared in /tmp/kitchen/cookbooks/postgresql/recipes/server_redhat.rb:54:in `from_file'

       template("/etc/sysconfig/pgsql/postgresql") do
         provider Chef::Provider::Template
         action "create"
         retries 0
         retry_delay 2
         guard_interpreter :default
         path "/etc/sysconfig/pgsql/postgresql"
         backup 5
         atomic_update true
         source "pgsql.sysconfig.erb"
         cookbook_name :postgresql
         recipe_name "server_redhat"
         mode "0644"
       end

       [2014-07-26T01:46:19+00:00] INFO: Running queued delayed notifications before re-raising exception

       Running handlers:
       [2014-07-26T01:46:19+00:00] ERROR: Running exception handlers
       Running handlers complete

       [2014-07-26T01:46:19+00:00] ERROR: Exception handlers complete
       [2014-07-26T01:46:19+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       Chef Client failed. 10 resources updated in 126.748339317 seconds
       [2014-07-26T01:46:19+00:00] ERROR: template[/etc/sysconfig/pgsql/postgresql] (postgresql::server_redhat line 54) had an error: Chef::Exceptions::EnclosingDirectoryDoesNotExist: Parent directory /etc/sysconfig/pgsql does not exist.
       [2014-07-26T01:46:19+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

**Comment by rmoriz** [comment]
I think the service name is still an issue when using PGDG packages on CentOS7.

e.g. 

```
           ================================================================================
           Error executing action `enable` on resource 'service[postgresql]'
           ================================================================================
           
           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           Expected process to exit with [0], but received '1'
           ---- Begin output of /bin/systemctl enable postgresql ----
           STDOUT: 
           STDERR: Failed to issue method call: No such file or directory
           ---- End output of /bin/systemctl enable postgresql ----
           Ran /bin/systemctl enable postgresql returned 1
           
```

IMHO the service name should be "postgresql-9.4". Chef should be able to do the systemd part, which is 

```sh
$ systemctl enable postgresql-9.4.service
ln -s '/usr/lib/systemd/system/postgresql-9.4.service' '/etc/systemd/system/multi-user.target.wants/postgresql-9.4.service'
```

**Comment by rmoriz** [comment]
CentOS still failing with PGDG releases, because the cluster init call is using "service". IMHO the code at https://github.com/hw-cookbooks/postgresql/blob/develop/recipes/server_redhat.rb#L74 needs to cover CentOS  >= 7, too.

Also there are some nasty "unless" calls and wierd things like https://github.com/hw-cookbooks/postgresql/blob/develop/recipes/server_redhat.rb#L80

I'm happy to provide a PR and fix this, however I already see related PR queued up (for example #230).

**Comment by ozzyogkush** [comment]
I am seeing the same error as described by `themasterchef` above at https://github.com/hw-cookbooks/postgresql/issues/167#issuecomment-54959580

See https://github.com/hw-cookbooks/postgresql/issues/245

**Comment by merqlove** [comment]
I think main mistake here:
[recipes/server_redhat.rb](hw-cookbooks/postgresql/recipes/server_redhat.rb)

I'm added this to develop branch. Seems that it works well.

- Primary here platform check problem (on CentOS it is `centos` instead of `redhat`).
- Secondary here postgres PATH problem.

```ruby
elsif platform?("centos") and node['platform_version'].to_i >= 7

  execute "/usr/pgsql-#{node['postgresql']['version']}/bin/postgresql#{node['postgresql']['version'].split('.').join}-setup initdb #{svc_name}" do
    not_if { ::FileTest.exist?(File.join(dir, "PG_VERSION")) }
  end
```

**Comment by ozzyogkush** [comment]
Any updates on the status of this? Is this going to be fixed? Are one of the aforementioned PRs going to be merged, or should I just fork it and try and fix it myself at this point?

**Comment by merqlove** [comment]
#269

**Comment by slyness** [comment]
Coming soon. #314

**Comment by slyness** [comment]
Resolved as of release 4.0.0 - https://github.com/hw-cookbooks/postgresql/releases/tag/v4.0.0
